"new keyword" warnings in C# generated code

This happens when compiling the generated code for `IceGrid/Admin` interface. Previously converted using `ice2slice`.

In this case the base interface `Glacier2::Session` doesn't define any operation with parameters, and so there is no generated `Request` in the base class.

```
  ice2slice-test succeeded with 4 warning(s) (2,3s) → bin/Debug/net8.0/ice2slice-test.dll
    /Users/jose/Documents/3.8/ice2slice-test/generated/IceGrid/Admin.IceRpc.cs(9790,29): warning CS0109: The member 'IAdminSessionService.Request' does not hide an accessible member. The new keyword is not required.
    /Users/jose/Documents/3.8/ice2slice-test/generated/IceGrid/Admin.IceRpc.cs(9955,29): warning CS0109: The member 'IAdminSessionService.Response' does not hide an accessible member. The new keyword is not required.
    /Users/jose/Documents/3.8/ice2slice-test/generated/IceGrid/Session.IceRpc.cs(474,29): warning CS0109: The member 'ISessionService.Request' does not hide an accessible member. The new keyword is not required.
    /Users/jose/Documents/3.8/ice2slice-test/generated/IceGrid/Session.IceRpc.cs(527,29): warning CS0109: The member 'ISessionService.Response' does not hide an accessible member. The new keyword is not required.
```

The logic is in:

https://github.com/icerpc/icerpc-csharp/blob/f109b98c1cb69755ccca7ddf3cc247982e7afb8c/tools/slicec-cs/src/generators/dispatch_generator.rs#L72-L86

We generate Request when an interface has operation with parameters. We are missing the same check for the base when adding the `new ` modifier.

--- a/tests/IceRpc.Slice.Tests/OperationTests.slice
+++ b/tests/IceRpc.Slice.Tests/OperationTests.slice
@@ -85,3 +85,10 @@ interface MyTaggedOperationsReadOnlyMemoryParams {
 interface MyBaseOperations {
     op()
 }
+
+interface NoOperation {}
+
+// Make sure the generated code does not generate a "new" for Request and Response in the service class.
+interface DerivedFromNoOperation : NoOperation {
+    op(x: int32) -> int32
+}

--- a/tools/slicec-cs/src/generators/dispatch_generator.rs
+++ b/tools/slicec-cs/src/generators/dispatch_generator.rs
@@ -67,8 +67,6 @@ private static readonly IActivator _defaultActivator =
 }
 
 fn request_class(interface_def: &Interface) -> CodeBlock {
-    let bases = interface_def.base_interfaces();
-
     let mut operations = interface_def.operations();
     operations.retain(|o| !o.parameters.is_empty());
 
@@ -76,8 +74,11 @@ fn request_class(interface_def: &Interface) -> CodeBlock {
         return "".into();
     }
 
+    let mut base_operations = interface_def.all_inherited_operations();
+    base_operations.retain(|o| !o.parameters.is_empty());
+
     let mut class_builder = ContainerBuilder::new(
-        if bases.is_empty() {
+        if base_operations.is_empty() {
             "public static class"
         } else {
             "public static new class"
@@ -143,8 +144,6 @@ fn request_class(interface_def: &Interface) -> CodeBlock {
 }
 
 fn response_class(interface_def: &Interface) -> CodeBlock {
-    let bases = interface_def.base_interfaces();
-
     let mut operations = interface_def.operations();
     operations.retain(|o| o.has_non_streamed_return_members());
 
@@ -152,8 +151,11 @@ fn response_class(interface_def: &Interface) -> CodeBlock {
         return "".into();
     }
 
+    let mut base_operations = interface_def.all_inherited_operations();
+    base_operations.retain(|o| o.has_non_streamed_return_members());
+
     let mut class_builder = ContainerBuilder::new(
-        if bases.is_empty() {
+        if base_operations.is_empty() {
             "public static class"
         } else {
             "public static new class"

